ddtrace/internal/coverage/instrumentation_py3_10.py

```diff
@@ -0,0 +1,312 @@
+        self.targets: t.List["Branch"] = []
+
+
+class Branch(ABC):
+    def __init__(self, start: Instruction, end: Instruction) -> None:
+        self.start = start
+        self.end = end
+
+    @property
+    def arg(self) -> int:
+        raise NotImplementedError
+
+
+class RBranch(Branch):
+    @property
+    def arg(self) -> int:
+        return abs(self.end.offset - self.start.offset - 2) >> 1
+
+
+class ABranch(Branch):
+    @property
+    def arg(self) -> int:
+        return self.end.offset >> 1
+
+
+EXTENDED_ARG = dis.EXTENDED_ARG
+
+
+def instr_with_arg(opcode: int, arg: int) -> t.List[Instruction]:
+    instructions = [Instruction(-1, opcode, arg & 0xFF)]
+    arg >>= 8
+    while arg:
+        instructions.insert(0, Instruction(-1, EXTENDED_ARG, arg & 0xFF))
+        arg >>= 8
+    return instructions
+
+
+def update_location_data(
+    code: CodeType, trap_map: t.Dict[int, int], ext_arg_offsets: t.List[t.Tuple[int, int]]
+) -> bytes:
+    # DEV: We expect the original offsets in the trap_map
+    new_data = bytearray()
+
+    data = code.co_linetable
+    data_iter = iter(data)
+    ext_arg_offset_iter = iter(sorted(ext_arg_offsets))
+    ext_arg_offset, ext_arg_size = next(ext_arg_offset_iter, (None, None))
+
+    original_offset = offset = 0
+    while True:
+        try:
+            if original_offset in trap_map:
+                # Give no line number to the trap instrumentation
+                trap_offset_delta = trap_map[original_offset] << 1
+                new_data.append(trap_offset_delta)
+                new_data.append(128)  # No line number
+                offset += trap_offset_delta
+
+            offset_delta = next(data_iter)
+            line_delta = next(data_iter)
+
+            original_offset += offset_delta
+            offset += offset_delta
+            if ext_arg_offset is not None and ext_arg_size is not None and offset > ext_arg_offset:
+                new_offset_delta = offset_delta + (ext_arg_size << 1)
+                new_data.append(new_offset_delta & 0xFF)
+                new_data.append(line_delta)
+                new_offset_delta >>= 8
+                while new_offset_delta:
+                    new_data.append(new_offset_delta & 0xFF)
+                    new_data.append(0)
+                    new_offset_delta >>= 8
+                offset += ext_arg_size << 1
+
+                ext_arg_offset, ext_arg_size = next(ext_arg_offset_iter, (None, None))
+            else:
+                new_data.append(offset_delta)
+                new_data.append(line_delta)
+        except StopIteration:
+            break
+
+    return bytes(new_data)
+
+
+LOAD_CONST = dis.opmap["LOAD_CONST"]
+CALL = dis.opmap["CALL_FUNCTION"]
+POP_TOP = dis.opmap["POP_TOP"]
+
+
+def trap_call(trap_index: int, arg_index: int) -> t.Tuple[Instruction, ...]:
+    return (
+        *instr_with_arg(LOAD_CONST, trap_index),
+        *instr_with_arg(LOAD_CONST, arg_index),
+        Instruction(-1, CALL, 1),
```

Comment:
Should we define `NO_OFFSET = -1` and then do
```suggestion
        Instruction(NO_OFFSET, CALL, 1),
```
to make this more readable?

Reply:
Can I offer to throw that in as a freebie in #9609 ?